src/test/java/uk/gov/ons/census/casesvc/messaging/UacCreatedEventReceiverIT.java

```diff
@@ -0,0 +1,95 @@
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+import org.springframework.transaction.annotation.Transactional;
+import uk.gov.ons.census.casesvc.model.dto.ResponseManagementEvent;
+import uk.gov.ons.census.casesvc.model.entity.Case;
+import uk.gov.ons.census.casesvc.model.entity.UacQidLink;
+import uk.gov.ons.census.casesvc.model.repository.CaseRepository;
+import uk.gov.ons.census.casesvc.model.repository.EventRepository;
+import uk.gov.ons.census.casesvc.model.repository.UacQidLinkRepository;
+import uk.gov.ons.census.casesvc.testutil.RabbitQueueHelper;
+
+@ContextConfiguration
+@ActiveProfiles("test")
+@SpringBootTest
+@RunWith(SpringJUnit4ClassRunner.class)
+@DirtiesContext(classMode = DirtiesContext.ClassMode.AFTER_EACH_TEST_METHOD)
+public class UacCreatedEventReceiverIT {
+
+  @Value("${queueconfig.rh-uac-queue}")
+  private String rhUacQueue;
+
+  @Value("${queueconfig.uac-qid-created-queue}")
+  private String uacQidCreatedQueue;
+
+  @Autowired private RabbitQueueHelper rabbitQueueHelper;
+  @Autowired private CaseRepository caseRepository;
+  @Autowired private EventRepository eventRepository;
+  @Autowired private UacQidLinkRepository uacQidLinkRepository;
+
+  @Before
+  @Transactional
+  public void setUp() {
+    rabbitQueueHelper.purgeQueue(uacQidCreatedQueue);
+    rabbitQueueHelper.purgeQueue(rhUacQueue);
+    eventRepository.deleteAllInBatch();
+    uacQidLinkRepository.deleteAllInBatch();
+    caseRepository.deleteAllInBatch();
+  }
+
+  @Test
+  public void testReceiveRmUacCreatedEvent() throws IOException, InterruptedException {
+    // Given
+    Case linkedCase = getRandomCase();
+    caseRepository.saveAndFlush(linkedCase);
+    BlockingQueue<String> uacEventQueue = rabbitQueueHelper.listen(rhUacQueue);
+
+    ResponseManagementEvent uacCreatedEvent = generateUacCreatedEvent(linkedCase);
+
+    // When
+    String uacCreatedEventJson = convertObjectToJson(uacCreatedEvent);
+    Message message =
+        MessageBuilder.withBody(uacCreatedEventJson.getBytes())
+            .setContentType(MessageProperties.CONTENT_TYPE_JSON)
+            .build();
+    rabbitQueueHelper.sendMessage(uacQidCreatedQueue, message);
+
+    // Then
+    // Check the UAC Updated event is emitted
+    ResponseManagementEvent uacUpdatedEvent =
+        rabbitQueueHelper.checkExpectedMessageReceived(uacEventQueue);
+    assertEquals(
+        uacCreatedEvent.getPayload().getUacQidCreated().getCaseId().toString(),
+        uacUpdatedEvent.getPayload().getUac().getCaseId());
+
+    // Check the Uac Qid Link is created
+    Optional<UacQidLink> actualUacQidLink =
+        uacQidLinkRepository.findByQid(uacCreatedEvent.getPayload().getUacQidCreated().getQid());
+    assertTrue(actualUacQidLink.isPresent());
+    assertEquals(
+        uacCreatedEvent.getPayload().getUacQidCreated().getUac(), actualUacQidLink.get().getUac());
+    assertEquals(linkedCase.getCaseId(), actualUacQidLink.get().getCaze().getCaseId());
+  }
+}
```

Comment:
As a team we need to decide whether we're going to always test @Transactional or not.  Some do it, some don't.  It's good to check that a failure will behave as expected (not ack msg and keep trying until it passes),  there's a transactionIT class which suppresses logging.

Reply:
@LukeLoze1 If we're going to require them then we need a tech session or something, I was not even aware they are a thing till now.

Reply:
yeap, they're quite easy - but we should just be consistent